sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +324,277 @@ public void run() {
       }
     }
   }
+
+  @Immutable
+  @AutoValue
+  public abstract static class Config {
+
+    public abstract boolean isReportOnlySampled();
+
+    public abstract long getScheduleDelayMillis();
+
+    public abstract int getMaxQueueSize();
+
+    public abstract int getMaxExportBatchSize();
+
+    public abstract int getExporterTimeoutMillis();
+
+    public static Config getDefault() {
+      return new Builder().build();
+    }
+
+    /**
+     * Creates a {@link Config} object reading the configuration values first from the environment
+     * and then from system properties. If a configuration value is missing, it uses the default
+     * value.
+     *
+     * @return The {@link Config} object.
+     */
+    public static Config loadFromDefaultSources() {
+      return new Builder().readEnv().readSystemProperties().build();
+    }
+
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public static final class Builder {
+
+      private static final String KEY_SCHEDULE_DELAY_MILLIS = "otel.bsp.schedule.delay";
+      private static final String KEY_MAX_QUEUE_SIZE = "otel.bsp.max.queue";
+      private static final String KEY_MAX_EXPORT_BATCH_SIZE = "otel.bsp.max.export.batch";
+      private static final String KEY_EXPORT_TIMEOUT_MILLIS = "otel.bsp.export.timeout";
+      private static final String KEY_SAMPLED = "otel.bsp.report.sampled";
+
+      private static final long DEFAULT_SCHEDULE_DELAY_MILLIS = 5000;
+      private static final int DEFAULT_MAX_QUEUE_SIZE = 2048;
+      private static final int DEFAULT_MAX_EXPORT_BATCH_SIZE = 512;
+      private static final int DEFAULT_EXPORT_TIMEOUT_MILLIS = 30_000;
+      private static final boolean DEFAULT_REPORT_ONLY_SAMPLED = true;
+
+      private long scheduleDelayMillis = DEFAULT_SCHEDULE_DELAY_MILLIS;
+      private int maxQueueSize = DEFAULT_MAX_QUEUE_SIZE;
+      private int maxExportBatchSize = DEFAULT_MAX_EXPORT_BATCH_SIZE;
+      private int exporterTimeoutMillis = DEFAULT_EXPORT_TIMEOUT_MILLIS;
+      private boolean sampled = DEFAULT_REPORT_ONLY_SAMPLED;
+
+      enum NamingConvention {
```

Comment:
So I feel like this could be re-used by other configuration logic. But of course, lets do that in a follow-up PR ;)